src/components/side_panel/chart/calendar_chart/calendar_chart_design_panel.ts

```diff
@@ -0,0 +1,67 @@
+import { Component } from "@odoo/owl";
+import { _t } from "../../../../translation";
+import { LegendPosition } from "../../../../types/chart";
+import { CalendarChartDefinition } from "../../../../types/chart/calendar_chart";
+import { Color, DispatchResult, SpreadsheetChildEnv, UID } from "../../../../types/index";
+import { SidePanelCollapsible } from "../../components/collapsible/side_panel_collapsible";
+import { RoundColorPicker } from "../../components/round_color_picker/round_color_picker";
+import { Section } from "../../components/section/section";
+import {
+  AxisDefinition,
+  AxisDesignEditor,
+} from "../building_blocks/axis_design/axis_design_editor";
+import { ColorScalePicker } from "../building_blocks/color_scale/color_scale_picker";
+import { GeneralDesignEditor } from "../building_blocks/general_design/general_design_editor";
+import { ChartShowValues } from "../building_blocks/show_values/show_values";
+
+interface Props {
+  chartId: UID;
+  definition: CalendarChartDefinition;
+  canUpdateChart: (chartId: UID, definition: Partial<CalendarChartDefinition>) => DispatchResult;
+  updateChart: (chartId: UID, definition: Partial<CalendarChartDefinition>) => DispatchResult;
+}
```

Comment:
This block is repeated many many times.
We should create something like this:
```typescript
export interface ChartSidePanelProps<T extends ChartDefinition> {
  chartId: UID;
  definition: T;
  canUpdateChart: (chartId: UID, definition: Partial<T>) => DispatchResult;
  updateChart: (chartId: UID, definition: Partial<T>) => DispatchResult;
}

export const ChartSidePanelPropsObject = {
  chartId: String,
  definition: Object,
  canUpdateChart: Function,
  updateChart: Function,
};
```

And use them everywhere.
I'm not sure of the naming